src/server/JasmineGraphServer.cpp

```diff
@@ -471,9 +470,33 @@ void JasmineGraphServer::resolveOperationalGraphs(){
 
 }
 
-//TODO:Needs to implement this method
 void JasmineGraphServer::deleteNonOperationalGraphFragment(int graphID) {
+    /**
+     * Method used in master node to commence deletion of a graph fragment
+     * INPUT:
+     *  graphID - ID of graph to be deleted
+     * OUTPUT: None
+     */
+    std::cout << "Deleting the graph fragments" << std::endl;
+    int count = 0;
+    //Define threads for each host
+    std::thread *deleteThreads = new std::thread[hostPortMap.size()];
+    //Iterate through all hosts
+    for (std::map<string, pair<int, int>>::iterator it = hostPortMap.begin(); it != hostPortMap.end(); it++) {
+        //Fetch hostname and port
+        string hostname = it -> first;
+        int port = (it -> second).first;
+        //Initialize threads for host
+        //Each thread runs the service to remove the given graph ID fragments in their datafolders
+        deleteThreads[count++] = std::thread(removeFragmentThroughService, hostname, port, to_string(graphID), this -> masterHost);
+        sleep(1);
+        server_logger.log("Deleted graph fragments of graph ID " + to_string(graphID), "info");
+    }
 
+    for (int threadCount = 0; threadCount < count; threadCount++) {
+        deleteThreads[threadCount].join();
```

Comment:
Here we should check whether the thread is joinable or not. See this [link](https://github.com/miyurud/jasminegraph/blob/master/src/server/JasmineGraphServer.cpp#L594).

Reply:
Similar to the line mentioned in the link, checked `deleteThreads[threadCount].joinable()` in an if block before main thread joins.